File: modules/security/exercises/05_pii_redact/solution/cpp/main.cpp
#include <iostream>
#include <regex>
#include <string>

static std::string redact_pii(const std::string& text) {
    return std::regex_replace(text, std::regex(R"([\w.-]+@[\w.-]+)"), "[EMAIL]");
}

int main() {
    std::string out = redact_pii("contact marco@example.com please");
    std::cout << (out.find("[EMAIL]") != std::string::npos ? "PII_REDACTED" : "LEAK") << "\n";
    return 0;
}

File: modules/security/exercises/05_pii_redact/solution/python/main.py
import re

def redact_pii(text: str) -> str:
    return re.sub(r"[\w.-]+@[\w.-]+", "[EMAIL]", text)

def main() -> None:
    log = "contact marco@example.com please"
    out = redact_pii(log)
    print("PII_REDACTED" if "[EMAIL]" in out else "LEAK")

if __name__ == "__main__":
    main()
